onebusaway-android/src/main/java/org/onebusaway/android/map/RouteMapController.java

```diff
@@ -211,6 +212,33 @@ public void onSaveInstanceState(Bundle outState) {
         outState.putString(MapParams.ROUTE_ID, mRouteId);
         outState.putBoolean(MapParams.ZOOM_TO_ROUTE, mZoomToRoute);
         outState.putBoolean(MapParams.ZOOM_INCLUDE_CLOSEST_VEHICLE, mZoomIncludeClosestVehicle);
+
+        Location centerLocation = mFragment.getMapView().getMapCenterAsLocation();
+        outState.putDouble(MapParams.CENTER_LAT, centerLocation.getLatitude());
+        outState.putDouble(MapParams.CENTER_LON, centerLocation.getLongitude());
+        outState.putFloat(MapParams.ZOOM, mFragment.getMapView().getZoomLevelAsFloat());
+        Log.v("save zoomlevel", mFragment.getMapView().getZoomLevelAsFloat() + "");
+    }
+
+    @Override
+    public void onViewStateRestored(Bundle savedInstanceState) {
+        String stopId = savedInstanceState.getString(MapParams.STOP_ID);
+        if (stopId == null) {
+            // If there is no focused stop then restore the map state otherwise
+            // let the BaseMapFragment to handle map state with focused stop
+
+            float mapZoom = savedInstanceState.getFloat(MapParams.ZOOM, MapParams.DEFAULT_ZOOM);
+            if (mapZoom != MapParams.DEFAULT_ZOOM) {
+                mFragment.getMapView().setZoom(mapZoom);
+            }
+
+            double lat = savedInstanceState.getDouble(MapParams.CENTER_LAT);
+            double lon = savedInstanceState.getDouble(MapParams.CENTER_LON);
+            if (lat != 0d && lon != 0d) {
+                Location location = LocationUtils.makeLocation(lat, lon);
+                mFragment.getMapView().setMapCenter(location, true, false);
```

Comment:
> I can't test this yet because of the NPE I commented on, but I'm thinking this should probably be:
> 
> mFragment.getMapView().setMapCenter(location, false, false);
> 
> ...so the map snaps to the location, rather than animates to it. Are we currently animating from 0.0 to this lat/long, or from a nearby lat/long to this lat/long?

I fixed this by using `mFragment.getMapView().setMapCenter(location, false, false);` . However, this only disables the animation when the sliding panel is not open. When it is visible, `HomeActivity` handles the orientation changes by calling `onArrivalTimesUpdated`:

```
public void onArrivalTimesUpdated(ObaArrivalInfoResponse response) {
        if (response == null || response.getStop() == null) {
            return;
        }

        // If we're missing any local references (e.g., if orientation just changed), store the values
        if (mFocusedStopId == null) {
            mFocusedStopId = response.getStop().getId();
        }
        if (mFocusedStop == null) {
            mFocusedStop = response.getStop();

            // Since mFocusedStop was null, the layout changed, and we should recenter map on stop
            if (mMapFragment != null && mSlidingPanel != null) {
                mMapFragment.setMapCenter(mFocusedStop.getLocation(), true,
                        mSlidingPanel.getPanelState() == SlidingUpPanelLayout.PanelState.ANCHORED);
            }

            // ...and we should add a focus marker for this stop
            if (mMapFragment != null) {
                mMapFragment.setFocusStop(mFocusedStop, response.getRoutes());
            }
        }

        // Header might have changed height, so make sure my location button is set above the header
        moveMyLocationButton();

        // Show arrival info related tutorials
        showArrivalInfoTutorials(response);
    }

```

And, this method animates the map. Therefor there is an inconsistency on rotation in `RouteMapController`. Map doesn't animate if the sliding panel is not open, but it animates when the panel is open.